Makefile

```diff
@@ -138,20 +138,20 @@ $(FFMPEG_ROOT):
 	git -C $(FFMPEG_ROOT) checkout release/6.1
 
 # For ARM64 builds, use the image ghcr.io/viamrobotics/antique:arm64 for backward compatibility
-FFMPEG_H264_PREFIX ?= $(shell realpath .)/gostream/codec/h264/ffmpeg/$(shell uname -s)-$(shell uname -m)
-ffmpeg-h264-static: $(FFMPEG_ROOT)
-	cd $(FFMPEG_ROOT) && ./configure \
-		--disable-programs \
-		--disable-doc \
-		--disable-everything \
-		--enable-encoder=h264_v4l2m2m \
-		--prefix=$(FFMPEG_H264_PREFIX) \
-		--enable-pic
+FFMPEG_PREFIX ?= $(shell realpath .)/gostream/ffmpeg/$(shell uname -s)-$(shell uname -m)
+FFMPEG_OPTS = --disable-programs --disable-doc --disable-everything --prefix=$(FFMPEG_PREFIX) --enable-pic --disable-autodetect
+ifeq ($(shell uname -m),aarch64)
+	FFMPEG_OPTS += --enable-encoder=h264_v4l2m2m
+	FFMPEG_OPTS += --enable-v4l2-m2m
+endif
+ffmpeg: $(FFMPEG_ROOT)
+	cd $(FFMPEG_ROOT) && $(MAKE) distclean
+	cd $(FFMPEG_ROOT) && ./configure $(FFMPEG_OPTS)
 	cd $(FFMPEG_ROOT) && $(MAKE)
 	cd $(FFMPEG_ROOT) && $(MAKE) install
 
-	# remove pkg-config and shared library files
-	rm -rf $(FFMPEG_H264_PREFIX)/lib/pkgconfig
-	rm -rf $(FFMPEG_H264_PREFIX)/share
+	# only keep archive files
+	find $(FFMPEG_PREFIX)/* -type d ! -wholename $(FFMPEG_PREFIX)/lib | xargs rm -rf
```

Comment:
why is this cleanup step necessary?

Reply:
`make install` will add source and pkg-config files to `FFMPEG_PREFIX`, and as far as I can tell, there's no way to prevent that. We don't need pkg-config files, and the source files are shared among the architectures and are in the directory above. I just added a comment too, to clarify.